pkg/systemd/services/timer-dialog.jsx

```diff
@@ -154,58 +265,77 @@ const CreateTimerDialogBody = ({ owner }) => {
                </>
            }>
             {dialogError && <ModalError dialogError={_("Timer creation failed")} dialogErrorDetail={dialogError} />}
-            <Form isHorizontal onSubmit={onSubmit}>
+            <Form isHorizontal id='timerForm' onSubmit={onSubmit} onChange={formCheck}>
                 <FormGroup label={_("Name")}
+                           isRequired
                            fieldId="servicename"
-                           validated={submitted && validationFailed.name ? "error" : "default"}
+                           validated={validationFailed.servicename ? "error" : "default"}
                            helperTextInvalid={!name.trim().length ? _("This field cannot be empty") : _("Only alphabets, numbers, : , _ , . , @ , - are allowed")}>
                     <TextInput id='servicename'
+                               isRequired
                                value={name}
-                               validated={submitted && validationFailed.name ? "error" : "default"}
+                               validated={validationFailed.servicename ? "error" : "default"}
+                               onBlur={validate}
                                onChange={setName} />
                 </FormGroup>
                 <FormGroup label={_("Description")}
+                           isRequired
                            fieldId="description"
-                           validated={submitted && validationFailed.description ? "error" : "default"}
+                           validated={validationFailed.description ? "error" : "default"}
                            helperTextInvalid={_("This field cannot be empty")}>
                     <TextInput id='description'
+                               isRequired
                                value={description}
-                               validated={submitted && validationFailed.description ? "error" : "default"}
+                               validated={validationFailed.description ? "error" : "default"}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)